resources/splunk_install.rb

```diff
@@ -0,0 +1,215 @@
+# Encoding: utf-8
+#
+# Cookbook Name:: cerner_splunk_ingredient
+# Resource:: splunk_install
+#
+# Resource for managing the installation of Splunk
+class SplunkInstall < ChefCompat::Resource
+  include CernerSplunk::PlatformHelpers, CernerSplunk::PathHelpers
+
+  resource_name :splunk_install
+
+  property :name, String, name_property: true, identity: true
+  property :package, [:splunk, :universal_forwarder], required: true
+  property :version, String, required: true
+  property :build, String, required: true
+  property :base_url, String, default: 'https://download.splunk.com/products'
+  property :accept_license, TrueClass, required: true
+
+  default_action :install
+
+  def after_created
+    node.run_state['splunk_ingredient'] ||= { 'installations' => {} }
```

Comment:
I wonder if that change would be better made later when there is actually another resource using the run state. It would be trivial to extract this into a helper method and remain passive later, and I can't determine the design for very future resources yet. It could be the case that the run state for every other resource exists under the context of an installation (because logically there might not be anything we're managing outside of installed Splunk).